cpp/include/raft/linalg/map_then_reduce.cuh

```diff
@@ -66,25 +67,27 @@ void mapThenSumReduce(
  * @param in the input array
  * @param args additional input arrays
  */
-
 template <typename InType,
           typename MapOp,
           typename ReduceLambda,
+          typename IdxType = std::uint32_t,
           int TPB          = 256,
           typename OutType = InType,
           typename... Args>
-void mapThenReduce(OutType* out,
-                   size_t len,
-                   OutType neutral,
-                   MapOp map,
-                   ReduceLambda op,
-                   cudaStream_t stream,
-                   const InType* in,
-                   Args... args)
+[[deprecated("Use function `mapReduce` from `raft/linalg/map_reduce.cuh")]] void mapThenReduce(
```

Comment:
Do you know if `breathe` / `doxygen` support this?

Reply:
Not sure but even Mark has used it in his `sampleWithReplacement` PR